js/values/Participle.js

```diff
@@ -0,0 +1,147 @@
+var defaults = require( "lodash/defaults" );
+var forEach = require( "lodash/forEach" );
+
+/**
+ * Default attributes to be used by the Participle if they are left undefined.
+ * @type { { auxiliaries: array, type: string } }
+ */
+var defaultAttributes = {
+	auxiliaries: [],
+	type: "",
+};
+/**
+ * Gets the parsed type name of attributes.
+ *
+ * @param {array|object|string|number} attribute The attribute to get the parsed type from.
+ * @returns {string} The parsed type name.
+ */
+var getType = function( attribute ) {
+	var rawTypeName = toString.call( attribute );
+	var parsedTypeName = "undefined";
+
+	switch( rawTypeName ) {
+		case "[object Array]":
+			parsedTypeName =  "array";
+			break;
+		case "[object Object]":
+			parsedTypeName = "object";
+			break;
+		case "[object String]":
+			parsedTypeName = "string";
+			break;
+		case "[object Number]":
+			parsedTypeName = "number";
+			break;
+		case "[object Boolean]":
+			parsedTypeName = "boolean";
+			break;
+		default:
+			return rawTypeName;
+	}
+	return parsedTypeName;
+};
+
+/**
+ * Construct the Participle object and set the participle, sentence part, auxiliary and type.
+ *
+ * @param {string} participle The participle.
+ * @param {string} sentencePart The sentence part where the participle is from.
+ * @param {object} attributes The object containing all attributes.
+ * @constructor
+ */
+var Participle = function( participle, sentencePart, attributes ) {
+	this._participle = participle || "";
+	this._sentencePart = sentencePart || "";
+	this._determinesSentencePartIsPassive = false;
+
+	if ( this._participle === "" ) {
+		throw Error( "The participle should not be empty." );
+	}
+
+	if ( this._sentencePart === "" ) {
+		throw Error( "The sentence part should not be empty." );
+	}
+
+	attributes = attributes || {};
+
+	this.validateAttributes( attributes );
+
+	defaults( attributes, defaultAttributes );
+	this._attributes = attributes;
+};
+
+/**
+ * Validates the type of all attributes. Throws an error if the type is invalid.
+ *
+ * @param {object} attributes The object containing all attributes.
+ * @returns {void}
+ */
+Participle.prototype.validateAttributes = function( attributes ) {
+	forEach( attributes, function( attributeValue, attributeName ) {
```

Comment:
The `closure` inside this function might become a standalone function as well. Something like `validateAttribute`